File: python-algorithm/leetcode/problem_1684.py
"""1684. Count the Number of Consistent Strings
https://leetcode.com/problems/count-the-number-of-consistent-strings/
"""
from typing import List


class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        allowed = set(allowed)
        ans = 0
        for word in words:
            flag = True
            for c in word:
                if c not in allowed:
                    flag = False
                    break
            if flag:
                ans += 1
        return ans

File: java-algo/src/main/java/com/leetcode/Problem1684.java
package com.leetcode;

import java.util.HashSet;
import java.util.Set;

/**
 * 1684. Count the Number of Consistent Strings
 * https://leetcode.com/problems/count-the-number-of-consistent-strings
 */
public class Problem1684 {
    public int countConsistentStrings(String allowed, String[] words) {
        int cnt = 0;
        Set<Character> set = new HashSet<>();
        for (char c : allowed.toCharArray()) {
            set.add(c);
        }
        for (String word : words) {
            for (char c : word.toCharArray()) {
                if (!set.contains(c)) {
                    cnt++;
                    break;
                }
            }
        }
        return words.length - cnt;
    }
}
